src/token/fa2.mligo

```diff
@@ -19,15 +19,18 @@ let validate_token_type (token_id, store : token_id * storage): token_id =
 
 [@inline]
 let check_sender (from_, token_id, store : address * nat * storage): address =
-  if (Tezos.sender = from_)
-  then from_
+  if token_id = store.frozen_token_id then
+    (failwith("FROZEN_TOKEN_NOT_TRANSFERABLE") : address)
   else
-    let key: operator = { owner = from_; operator = sender; token_id = token_id } in
-    if Big_map.mem key store.operators
+    if (Tezos.sender = from_)
     then from_
     else
-     ([%Michelson ({| { FAILWITH } |} : string * unit -> address)]
-        ("FA2_NOT_OPERATOR", ()) : address)
+      let key: operator = { owner = from_; operator = sender; token_id = token_id } in
+      if Big_map.mem key store.operators
+      then from_
+      else
+        ([%Michelson ({| { FAILWITH } |} : string * unit -> address)]
+            ("FA2_NOT_OPERATOR", ()) : address)
```

Comment:
The logic seems correct to me, but I think this will be a little more readable (and maybe even more efficient) if implemented as:
```ocaml
let check_sender (from_, token_id, store : address * nat * storage): address =
  if token_id = store.frozen_token_id
  then (failwith("FROZEN_TOKEN_NOT_TRANSFERABLE") : address)
  else if (Tezos.sender <> from_)
    let key: operator = { owner = from_; operator = sender; token_id = token_id } in
    if Big_map.mem key store.operators
    then from_
    else ([%Michelson ({| { FAILWITH } |} : string * unit -> address)] 
            ("FA2_NOT_OPERATOR", ()) : address)
  else from_
```